spec.html

```diff
@@ -7579,58 +7579,43 @@ <h1>Jobs and Job Queues</h1>
+          <li>Perform any implementation-defined cleanup steps.</li>
+          <li>Pop the previously-pushed execution context from the execution context stack.</li>
+        </ol>
+      </li>
+      <li>Only one Job may be actively undergoing evaluation at any point in time.</li>
+      <li>Once evaluation of a Job starts, it must run to completion before evaluation of any other Job starts.</li>
+      <li>The abstract operation must return a normal completion, implementing its own handling of errors.</li>
+    </ul>
 
-    <emu-clause id="sec-enqueuejob" aoid="EnqueueJob">
-      <h1>EnqueueJob ( _queueName_, _job_, _arguments_ )</h1>
-      <p>The EnqueueJob abstract operation requires three arguments: _queueName_, _job_, and _arguments_. It performs the following steps:</p>
+    <emu-clause id="sec-hostenqueuepromisejob" aoid="HostEnqueuePromiseJob">
+      <h1>HostEnqueuePromiseJob ( _pending_ )</h1>
+
+      <p>HostEnqueuePromiseJob is a host-defined abstract operation that schedules the PendingJob record _pending_ to be performed, at some future time. The PendingJob records used with this algorithm are intended to be related to the handling of Promises, or otherwise, to be scheduled with equal priority to Promise handling operations.</p>
+
+      <p>The implementation of HostEnqueuePromiseJob must conform to the requirements in <emu-xref href="#sec-jobs-and-job-queues"></emu-xref>. Additionally, Jobs must be scheduled in FIFO order, with Jobs running in the same order as the HostEnqueuePromiseJob invocations which scheduled them.</p>
+    </emu-clause>
+
+    <emu-clause id="sec-enqueuepromisejob" aoid="EnqueuePromiseJob">
+      <h1>EnqueuePromiseJob ( _job_, _arguments_ )</h1>
+      <p>The EnqueuePromiseJob abstract operation requires two arguments: _job_ and _arguments_. It performs the following steps:</p>
       <emu-alg>
-        1. Assert: Type(_queueName_) is String and its value is the name of a Job Queue recognized by this implementation.
         1. Assert: _job_ is the name of a Job.
         1. Assert: _arguments_ is a List that has the same number of elements as the number of parameters required by _job_.
         1. Let _callerContext_ be the running execution context.
         1. Let _callerRealm_ be _callerContext_'s Realm.
         1. Let _callerScriptOrModule_ be _callerContext_'s ScriptOrModule.
         1. Let _pending_ be PendingJob { [[Job]]: _job_, [[Arguments]]: _arguments_, [[Realm]]: _callerRealm_, [[ScriptOrModule]]: _callerScriptOrModule_, [[HostDefined]]: *undefined* }.
-        1. Perform any implementation or host environment defined processing of _pending_. This may include modifying the [[HostDefined]] field or any other field of _pending_.
-        1. Add _pending_ at the back of the Job Queue named by _queueName_.
+        1. Let _result_ be the result of performing HostEnqueuePromiseJob using &laquo; _pending_ &raquo; as its arguments.
```

Comment:
```suggestion
        1. Let _result_ be the result of performing HostEnqueuePromiseJob with _pending_ as the argument.
```

I thought there was an issue where we discussed normalising this notation, but I can't find it right now...

Reply:
I'd expect it to say `Let _result_ be HostEnqueuePromiseJob(_pending_)` with the appropriate ?/! prefix, rather than being prose.

Reply:
Yeah, I prefer @ljharb 's preferred wording (I'd use ! here). I can't remember why it isn't already written like that.